publish-hook/src/main/java/com/redhat/lightblue/hook/publish/PublishHook.java

```diff
@@ -133,25 +127,29 @@ public void processHook(EntityMetadata entityMetadata, HookConfiguration hookCon
         return rootIdentities;
     }
 
-    private void insert(String entityName, Object entity) throws ClassNotFoundException, IllegalAccessException, InvocationTargetException, IOException,
-            NoSuchMethodException, InstantiationException {
+    private void insert(String entityName, Object entity) {
         ObjectNode jsonNode = new ObjectNode(JsonNodeFactory.instance);
         jsonNode.put("entity", entityName);
         ArrayNode data = jsonNode.putArray("data");
         data.add(new ObjectMapper().valueToTree(entity));
 
         InsertionRequest ireq = InsertionRequest.fromJson(jsonNode);
-        Response r = lightblueFactory.getMediator().insert(ireq);
+        Response r;
+        try {
+            r = lightblueFactory.getMediator().insert(ireq);
+        } catch (ClassNotFoundException | IllegalAccessException | InvocationTargetException | NoSuchMethodException | InstantiationException | IOException e) {
+            throw Error.get(HOOK_NAME + ":MediatorException", e);
+        }
         if (!r.getErrors().isEmpty()) {
-            // TODO Better Handle Exception
             for (Error e : r.getErrors()) {
-                LOGGER.error(e.toString());
+                LOGGER.error("Lightblue Error", e);
             }
+            throw Error.get(HOOK_NAME + ":Error", "Errors while attempting to insert esb events.");
```

Comment:
Mediator catches all exceptions at the end, and anything that's not handled
ends up at the client side as a generic error.

On Wed, Sep 30, 2015 at 7:28 AM, Naveen Malik <notifications@github.com>
wrote:

> In
> publish-hook/src/main/java/com/redhat/lightblue/hook/publish/PublishHook.java
> <https://github.com/lightblue-platform/lightblue-esb-hook/pull/37#discussion_r40792573>
> :
>
> >              }
> > +            throw Error.get(HOOK_NAME + ":Error", "Errors while attempting to insert esb events.");
>
> @bserdar <https://github.com/bserdar> does the hook framework protect
> against implementations throwing unchecked exception?
> … <#1501e71126211ccb_>
> On Wed, Sep 30, 2015 at 9:27 AM, Burak Serdar <notifications@github.com>
> wrote: In
> publish-hook/src/main/java/com/redhat/lightblue/hook/publish/PublishHook.java
> <
> https://github.com/lightblue-platform/lightblue-esb-hook/pull/37#discussion_r40792451>
> : > } > + throw Error.get(HOOK_NAME + ":Error", "Errors while attempting to
> insert esb events."); Agree with Naveen. Log errors, don't throw. — Reply
> to this email directly or view it on GitHub <
> https://github.com/lightblue-platform/lightblue-esb-hook/pull/37/files#r40792451>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-esb-hook/pull/37/files#r40792573>
> .
>